main/src/main/scala/sbt/internal/CrossJava.scala

```diff
@@ -367,11 +369,42 @@ private[sbt] object CrossJava {
           }
     }
 
+    class WindowsDiscoverConfig extends JavaDiscoverConf {
+      val base: File = file("C://Program Files/Java")
```

Comment:
You can use `sys.env.get("ProgramFiles")`.

In 64-bit machine, `%ProgramFiles%` points to:
- `C:\Program Files` if running sbt with 64-bit Java
- `C:\Program Files (x86)` if running sbt with 32-bit Java

In 32-bit machine, `%ProgramFiles%` points to `C:\Program Files`.

Reply:
That variable isn't filled for me (sbt 1.2.1 on cygwin on windows8)?

Should we just look in  `C:\Program Files (x86)/Java` as well as `C:\Program Files/Java`?

Reply:
Updated the PR to look in both.

To follow up on my last comment: the environment variable is `%PROGRAMFILES%` (case-sensitive), which on 32-bits machines points to `C:\Program Files` and on 64-bits machines points to `C:\Program Files` or `C:\Program Files (x86)` depends on your current process (https://stackoverflow.com/questions/17688758/using-programfilesx86-on-windows-os-32bit).

Can we invoke 32-bit Java from 64-bit sbt and vice-versa?

Reply:
> Can we invoke 32-bit Java from 64-bit sbt and vice-versa?

I am not sure if we tested specifically for these scenarios, but I'm guessing it's not too different from calling JDK 11 from JDK 8: You invoke java.exe in another process, and hope that serialization still hold up.

Reply:
Right: then checking for both installations seems to be the reasonable thing to do (even if it might then pick the 'wrong' architecture)